src/renderer/vt/XtermEngine.hpp

```diff
@@ -54,9 +54,18 @@ namespace Microsoft::Console::Render
         [[nodiscard]] HRESULT WriteTerminalW(const std::wstring_view str) noexcept override;
 
     protected:
+        // I'm using a non-class enum here, so that the values
+        // are trivially convertible and comparable to bool.
+        enum Tribool : uint8_t
+        {
+            Tribool_False = 0,
+            Tribool_True,
+            Tribool_Invalid,

```

Comment:
invalid is truthy! that makes sense for this case, but it is different from an `optional<bool>` and SQL's nullable bool

should we rename it to `CursorState` and be very specific on the inner constant names?

Reply:
Yea that's kinda my takeaway as well - Either this should be just a plain old optional or an enum, right?

Reply:
I mean it is an enum, I just wanted to reduce the amount of code I'm refactoring for... reasons. Ya know.
I'll make it an `enum class` then.

Reply:
Turns out that this was really simple... I removed the assertions from the unit test, since we can already observe the correctness of the algorithm by the way it emits escape sequences to us.